sdk/appconfiguration/azappconfiguration/client_responses.go

```diff
@@ -0,0 +1,102 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azappconfiguration
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+
+	"sdk/appconfiguration/sdk/appconfiguration/azappconfiguration/internal/generated"
+)
+
+type ConfigurationSettingResult struct {
+	ConfigurationSetting
+	ETag      *azcore.ETag
+	SyncToken *string
+}
+
+type configurationSettingResponse struct {
+	Result      ConfigurationSettingResult
+	RawResponse *http.Response
+}
+
+func responseFromGeneratedPut(g generated.AzureAppConfigurationClientPutKeyValueResponse) configurationSettingResponse {
+	return configurationSettingResponse{
+		Result: ConfigurationSettingResult{
+			ConfigurationSetting: configurationSettingFromGenerated(g.KeyValue),
+			ETag:                 (*azcore.ETag)(g.Etag),
+			SyncToken:            g.SyncToken,
+		},
+		RawResponse: g.RawResponse,
+	}
+}
+
+func responseFromGeneratedDelete(g generated.AzureAppConfigurationClientDeleteKeyValueResponse) configurationSettingResponse {
+	return configurationSettingResponse{
+		Result: ConfigurationSettingResult{
+			ConfigurationSetting: configurationSettingFromGenerated(g.KeyValue),
+			ETag:                 (*azcore.ETag)(g.Etag),
+			SyncToken:            g.SyncToken,
+		},
+		RawResponse: g.RawResponse,
+	}
+}
+
+func responseFromGeneratedPutLock(g generated.AzureAppConfigurationClientPutLockResponse) configurationSettingResponse {
+	return configurationSettingResponse{
+		Result: ConfigurationSettingResult{
+			ConfigurationSetting: configurationSettingFromGenerated(g.KeyValue),
+			ETag:                 (*azcore.ETag)(g.Etag),
+			SyncToken:            g.SyncToken,
+		},
+		RawResponse: g.RawResponse,
+	}
+}
+
+func responseFromGeneratedDeleteLock(g generated.AzureAppConfigurationClientDeleteLockResponse) configurationSettingResponse {
+	return configurationSettingResponse{
+		Result: ConfigurationSettingResult{
+			ConfigurationSetting: configurationSettingFromGenerated(g.KeyValue),
+			ETag:                 (*azcore.ETag)(g.Etag),
+			SyncToken:            g.SyncToken,
+		},
+		RawResponse: g.RawResponse,
+	}
+}
+
+type GetConfigurationSettingResult struct {
+	ConfigurationSettingResult
+	LastModified *time.Time
+}
+
+type GetConfigurationSettingResponse struct {
+	Result      GetConfigurationSettingResult
+	RawResponse *http.Response
+}
```

Comment:
I think we can merge these two types. It will reduce the number of exported models and make it more friendly for customers. Thoughts @jhendrixMSFT ?

Reply:
Jeffrey has told me that we don't need to have the `RawResponse` anymore (there should be ways to get the response some other way), so I merged Response into Type, but I'll check with @jhendrixMSFT